connectors/sources/github.py

```diff
@@ -1371,50 +1528,50 @@ async def _get_repo_object_for_github_app(self, owner, repo_name):
 
         return cached_repo[owner][full_repo_name]
 
-    async def _get_invalid_repos_for_personal_access_token(self):
+    async def _get_invalid_repos_for_personal_access_token(self) -> list[str]:
         try:
+            # Combine all repos for unified batch validation
             if self.configuration["repo_type"] == "other":
                 logged_in_user = await self._logged_in_user()
                 foreign_repos, configured_repos = self.github_client.bifurcate_repos(
                     repos=self.configured_repos,
                     owner=logged_in_user,
                 )
-                if logged_in_user not in self.user_repos:
-                    self.user_repos[logged_in_user] = {}
-                async for repo in self.github_client.get_user_repos(logged_in_user):
-                    self.user_repos[logged_in_user][repo["nameWithOwner"]] = repo
-                invalid_repos = list(
-                    set(configured_repos) - set(self.user_repos[logged_in_user].keys())
-                )
-
-                for repo_name in foreign_repos:
-                    try:
-                        self.foreign_repos[
-                            repo_name
-                        ] = await self.github_client.get_foreign_repo(
-                            repo_name=repo_name
-                        )
-                    except Exception:
-                        self._logger.debug(f"Detected invalid repository: {repo_name}.")
-                        invalid_repos.append(repo_name)
+                all_repos = configured_repos + foreign_repos
             else:
                 foreign_repos, configured_repos = self.github_client.bifurcate_repos(
                     repos=self.configured_repos,
                     owner=self.configuration["org_name"],
                 )
-                configured_repos.extend(foreign_repos)
-                if self.configuration["org_name"] not in self.org_repos:
-                    self.org_repos[self.configuration["org_name"]] = {}
-                async for repo in self.github_client.get_org_repos(
-                    self.configuration["org_name"]
-                ):
-                    self.org_repos[self.configuration["org_name"]][
-                        repo["nameWithOwner"]
-                    ] = repo
-                invalid_repos = list(
-                    set(configured_repos)
-                    - set(self.org_repos[self.configuration["org_name"]].keys())
+                all_repos = configured_repos + foreign_repos
+
+            invalid_repos = []
+            if all_repos:
```

Comment:
Do we need the if? If it's empty won't `self.github_client.get_repos_by_fully_qualified_name_batch` return empty dict?

Reply:
Indeed. Nice, thanks.